icu4j/main/classes/core/src/com/ibm/icu/text/CharsetRecog_ASCII.java

```diff
@@ -0,0 +1,39 @@
+// © 2016 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+package com.ibm.icu.text;
+
+/**
+ * Charset recognizer for plain ASCII
+ */
+class CharsetRecog_ASCII extends com.ibm.icu.text.CharsetRecognizer {
+
+    @Override
+    String getName() {
+        return "ASCII";
+    }
+
+    /* (non-Javadoc)
+     * @see com.ibm.icu.text.CharsetRecognizer#match(com.ibm.icu.text.CharsetDetector)
+     */
+    @Override
+    com.ibm.icu.text.CharsetMatch match(com.ibm.icu.text.CharsetDetector det) {
+        // code similar to determining com.ibm.icu.text.CharsetDetector.fC1Bytes
+        boolean highestBitSet = false;
+        for (int i = 0x80; i <= 0xFF; i += 1) {
+            if (det.fByteStats[i] != 0) {
+                highestBitSet = true;
+                break;
+            }
+        }
+
+        if (highestBitSet) {
+            // non-ASCII, because (at least) one byte in the stream is >= 128
+            return null;
+        } else {
+            // ASCII, because ALL bytes in the stream are <= 127.
+            // However, there could be some unicode (such as Hebrew) which also has this property.
```

Comment:
Hm? It could be some unusual charset like UTF-7 or [HZ](https://en.wikipedia.org/wiki/HZ_(character_encoding)), but [those are "prohibited character encodings" in modern HTML](https://html.spec.whatwg.org/#character-encodings) and have generally fallen out of favor.

I also don't think that Hebrew is relevant here.

Seems like the confidence for ASCII should be high if all bytes are 00..7F.